src/legacy/server/saved_objects/saved_objects_mixin.js

```diff
@@ -136,6 +137,7 @@ export function savedObjectsMixin(kbnServer, server) {
     types: visibleTypes,
     SavedObjectsClient,
     SavedObjectsRepository,
+    createNamespace: id => new Namespace(id),
```

Comment:
I'll defer to @skaapgif on this, but using a `createNamespace` function instead of returning the `Namespace` class differs from what we're doing above for both the `SavedObjectsClient` and the `SavedObjectsRepository` classes/constructors. Was there any specific reasoning for you differing from the previous precedent?

Reply:
I feel like we have both precedents defined here: we have the classes defined above, but then factory functions disguised as getters below. I'm happy to go in either direction, but I don't know that one is more consistent than the other

Reply:
We definitely do both approaches, which also bothers me. The tie-breaker for me was that if we could do static imports from OSS into X-Pack, I don't think we would've introduced a `createNamespace` factory, we would've just import and use the `Namespace` class. Where-as we would've introduced a `getSavedObjectsRepository` factory function somewhere because it's doing quite a lot.

Reply:
> We definitely do both approaches, which also bothers me. The tie-breaker for me was that if we could do static imports from OSS into X-Pack, I don't think we would've introduced a createNamespace factory, we would just import and use the Namespace class. Where-as we would've introduced a getSavedObjectsRepository factory function somewhere because it's doing quite a lot.

I'll buy that argument